src/api/supportRemindersApi.ts

```diff
@@ -0,0 +1,22 @@
+import { isProd } from '../lib/env';
+
+type ReminderPlatform = 'WEB' | 'AMP' | 'MMA' | 'SUPPORT';
+
+type ReminderComponent = 'EPIC' | 'BANNER' | 'THANKYOU' | 'CANCELLATION';
+
+type ReminderStage = 'PRE' | 'POST' | 'WINBACK';
+
+export interface BaseSignupRequest {
+    email: string;
+    reminderPlatform: ReminderPlatform;
+    reminderComponent: ReminderComponent;
+    reminderStage: ReminderStage;
+    reminderPeriod: string;
+    country?: string;
+    reminderOption?: string;
+}
```

Comment:
It might be worth narrowing down the types here? As far as dotcom-components is concerned the only valid platforms are `'WEB' | 'AMP'`, components `'EPIC' | 'BANNER'` and stage `'PRE' | 'POST'`. Also maybe we should include the `OneOffSignupRequest` like in `support-reminders`. `BaseSignupRequest` doesn't sound like something that should be used directly

Reply:
If `OneOffSignupRequest` inherits from `BaseSignupRequest`, what keys do we need to add to it?

Could you point me to the implementation of this on support-reminders @TomPretty?

Reply:
Yeah sure thing! It's over here: https://github.com/guardian/support-reminders/blob/main/src/create-reminder-signup/lambda/models.ts#L52

The only difference is that the `reminderPeriod` would be pulled out of this one and added to the `OneOffSignupRequest`